approach.tex

```diff
@@ -29,19 +29,6 @@ \subsection{Managing and Reporting Test Execution}
 This information enables us to generate reports on the execution of each test\footnote{These test reports may, on occasion, be issued as Word or \LaTeX{} documents, but this is not, in general, required.}, and ultimately to build a Verification Control Document (VCD; see \figref{fig:doctree}).
 The VCD will provide the verification status of each DM requirement (in terms of the fraction of test cases pertaining to that requirement which have been successfully executed).
 
-
```

Comment:
The diagram try to show the different test documents and how they a related each-other. 
My suggestion is to keep it in.

Reply:
I agree that the document shall not include too much technical details. The confluence page I have included as a reference is very detailed already, and it would be good to move that page in a technical note.

In general I agree with all your changes, except the removal of the validation document tree overview, that gives in my opinion a schema to read and global document tree, and a schematic understanding of DM itself.

Shall I created a branch from your branch in order to complete the few changes you are asking?
Or should I wait you merge your changes and create a new branch from master?

Follows few considerations on your comments.

2.2

The content of table 1 has been improved to be more coherent with the defined product tree. Yes there is a discrepancy, but this should be solved clarifying the meaning of “primary components”  as it was specified in the first bullet. As far as I understood we don’t havre such a classification in LDM-148 not in LDM-294.

I agree that if we say `milestones shall specify which components are impacted`, we shall do that before issuing the document.

2.3.2

Strictly speaking the source of true is in MD, so change control is applied to MD content and then documents are produced, reviewed and baselined in Docushare. However I am OK with the refactored text.

From a Data Management perspective, the synchronization back to MD is not relevant. In my understanding, we will base the DM VCD only on Jira information.

OK, I will add the following link:
https://confluence.lsstcorp.org/display/SYSENG/LSST+Verification+Architecture
I don't think SE has more documentation available.

2.3.3

Section 2.1 need to be updated, this is a remaining from the previous version of the document.

Introducing the Jira verification approach, test reports need to be baselined, as prove of the requirement verification.

More than views are folders.

If you prefer, we can add an additional document type, just to document the plan. But I think the effort to combine the two Jira elements is a better solution.

2.3.4

Yes, each test case will be baselined only in one test specification document.

2.3.5

Only the last execution time is reported in the VCD. We can list all execution times (and DMTRs), but this list can potentially be long and repeated many times in the table. 

I will fix the table spacing, this is quite tedious and time consuming.
The font size is made in order to put in evidence the requirement and the corresponding V. Element, test case and execution.

2.4

Regarding the test case owner, I don’t see any problem.
The test specification is approved via RFC (or LCR) and the test report is approved by who requested the test, that in general should not be the product owner.
The test case owner is in charge to write and maintain the test case.
Adding this sentence may clarify the text.

No. The SPR here are issues in the software under test, that need to be fixed. A deviation is a request to deviate from the baselined requirement (verification element).
If a SPR can’t be resolved, a Deviation (request for) on the corresponding requirement shall be opened.

Probably the best person to assess the SPR is the product owner. The corresponding CCB and who requested the test activity, may be involved in the deviation process, in case the SPR can’t be resolved. 
However this is still under definition by the SE.
I suggest to keep the text as is for now.

2.5

The scope of the pass/fail criteria is the test case.
The scope of a deviation is a verification element and therefore the corresponding requirement.
I suggest to keep the definition separated.

Reply:
> Shall I created a branch from your branch in order to complete the few changes you are asking?  Or should I wait you merge your changes and create a new branch from master?

I suggest you branch off this branch and make whatever changes you are planning.

> > 2. Verification Tests
> > 
> > 2.2 Components Under Test
> > 
> > Is there a distinction between “product” and “component”? We suddenly change
> > from the former to the latter here.

(Still needs to be addressed.)

> > The relationship of table 1 to the contents of LDM-148 is not as direct as
> > I'd like: not everything in LDM-148 appears in this table (e.g. the OODS,
> > but there are many other examples). Maybe that's because we still have work
> > to do to get the product tree right.
> 
> The content of table 1 has been improved to be more coherent with the
> defined product tree.

I'm not sure where these improvements have taken place — can you show me
where to find the improved version?

> Yes there is a discrepancy, but this should be solved clarifying the
> meaning of “primary components” as it was specified in the first
> bullet. As far as I understood we don’t havre such a classification in
> LDM-148 not in LDM-294.

I'm not sure why we don't just use the same list of components
everywhere, instead of making some “primary”.

> > As of now, the milestones don't describe which components they involve. We
> > should add this before baselining a version of the document which says they
> > do.
> 
> I agree that if we say milestones shall specify which components are
> impacted, we shall do that before issuing the document.

Ok.
  
> > 2.3.2 Requirements and Test Objects
> > 
> > Is change control applied to requirements in MagicDraw, or to documents
> > (like LSE-61) which are effectively snapshots of parts of MagicDraw? I think
> > the latter and have edited the text to say that, but I might be wrong.
> 
> Strictly speaking the source of true is in MD,

Does this mean that the CCB will approve all changes to MagicDraw rather
than to baselined documents? That seems very surprising! How will you
distinguish between which MagicDraw changes require Project-level CCB
control and which are covered by the DM-CCB?

> so change control is applied to MD content and then documents are
> produced, reviewed and baselined in Docushare. However I am OK with
> the refactored text.

I think if the text is not accurate we should change it to something
which is. It's very confusing to work out where change control is
actually being applied: let's make sure that this document provides an
accurate an unambiguous definition.

> > What's the process for syncing updates made in Jira back to MagicDraw? The
> > text says “System Engineering will synchronize them with MagicDraw once the
> > update is complete”, but how to SE know when an update is happening? What
> > does the product owner actually do? (This would actually be more
> > appropriate in a user mnanual than in this document).
> 
> From a Data Management perspective, the synchronization back to MD is
> not relevant. In my understanding, we will base the DM VCD only on Jira
> information.

If it's not relevant, let's remove all mention of it.

> > I don't understand the distinction between “test case” and “test plan” based
> > on the definitions in this section. In particular, I don't understand the
> > assertion that a test plan does not relate to a test case.
> > 
> > Actually, I found
> > https://www.adaptavist.com/doco/display/KT/Terms+and+Concepts which seems to
> > be clearer than some of the definitions here; I've made some changes based
> > on that.
> > 
> > Please provide a link to the systems engineering documentation that this
> > section refers to.
> 
> OK, I will add the following link:
> https://confluence.lsstcorp.org/display/SYSENG/LSST+Verification+Architecture
> I don't think SE has more documentation available.

That page shows the workflows as promised, so I think it is good enough.

> > 2.3.3. Test documents
> > 
> > This section says that DMTRs must be placed in Docushare and added to the
> > baseline. However, §2.1 says “test reports may, on occasion, be issued as
> > Word or LATEX documents, but this is not, in general, required”. This seems
> > to be contradictory.
> 
> Section 2.1 need to be updated, this is a remaining from the previous
> version of the document.

Ok, let's make sure it gets updated then.

> > In general (and referring to the previous section), we should be
> > careful about what is ACTUALLY baselined. I believe that the baseline
> > consists of the documentation which is approved by the CCB. Jira and
> > MagicDraw might provide a convenient interface for updating and
> > regenerating that documentation, but the requirements there are not,
> > themselves, baselined.  Right?
> 
> Introducing the Jira verification approach, test reports need to be
> baselined, as prove of the requirement verification.

I don't believe a test report can be part of the baseline. Per (e.g.)
https://www.sebokwiki.org/wiki/Baseline_(glossary), the baseline defines
a specification or product. A test report is descriptive, rather than
normative, by definition.

> > This section reverts back to talking about “products” rather than
> > components.
> > 
> > I'm not sure what it means for things to be available as “views” in Jira.
> > Are these like filters?
> 
> More than views are folders.
  
I don't know what this means.

> > I've removed a lot of the text from this section. I don't think it's
> > appropriate to provide tutorial style information in this document: that
> > would be better on Confluence.
> > 
> > Why are test plan and report documents combined into one? This seems to make
> > everything harder, from writing about them (it's a lot of words) to
> > approving them (see the complicated procedure in the next section).
> 
> If you prefer, we can add an additional document type, just to document
> the plan. But I think the effort to combine the two Jira elements is a
> better solution.

It seems like having two document types would make the procedures
easier, but I don't feel strongly about this as long as it is clearly
documented.
  
> > 2.3.4 Approval Procedure
> > 
> > I think the approval procedure only really works if each test case can
> > appear in only one test specification. Is that correct?
> 
> Yes, each test case will be baselined only in one test specification document.

Ok.

> > Note approval of test specs may not be by RFC (since it may need to go to
> > the project CCB). I have changed the text.
> > 
> > 2.3.5 The DM VCD
> > 
> > I don't understand why we list a single “run date” for test cases in the VCD
> > listed. Surely each test case will be run multiple times? What is this “run
> > date” supposed to represent?
> 
> Only the last execution time is reported in the VCD. We can list all
> execution times (and DMTRs), but this list can potentially be long and
> repeated many times in the table.

I think that's fine, but you need to say this in the text.
 
> > The formatting of this VCD example is crazy — why is everything different
> > sizes? Why is the spacing inconsistent? I was going to fix it, but since I
> > don't understand it I don't think I can do so properly.
> 
> I will fix the table spacing, this is quite tedious and time consuming.
> The font size is made in order to put in evidence the requirement and
> the corresponding V. Element, test case and execution.

Thank you for fixing this. Please make it stylistically consistent with
the other tables in the document.
  
> > 2.4 Roles and Personnel
> > 
> > The role of “owner” as described here appears to be in conflict with §2.3.4.
> > Specifically:
> > 
> > I expect the owner to be the product owner for the product being tested;
> > I expect that product owner to approve test reports;
> > If that owner were also the tester, they would be approving their own
> > reports, which should not happen.
> > I suggest therefore that the owner is not, in general, the tester.
> 
> Regarding the test case owner, I don’t see any problem.  The test
> specification is approved via RFC (or LCR) and the test report is
> approved by who requested the test, that in general should not be the
> product owner.  The test case owner is in charge to write and maintain
> the test case.  Adding this sentence may clarify the text.

Ok, if you can clarify the text here that would be helpful.
  
> > Are the Software Problem Reports described in this section the same as the
> > Deviations, described earlier?
> 
> No. The SPR here are issues in the software under test, that need to be
> fixed. A deviation is a request to deviate from the baselined
> requirement (verification element).  If a SPR can’t be resolved, a
> Deviation (request for) on the corresponding requirement shall be
> opened.

Ok, this is certainly not clear from the text as written.

> > Per §2.3.4, a test campaign may be requested by anybody. It's not practical
> > to suggest that the DM-CCB will assess and approve SPRs from potentially
> > infinite test campaigns. This procedure can't work as described. I guess you
> > really want the person who requested the test to assess SPRs?
> 
> Probably the best person to assess the SPR is the product owner. The
> corresponding CCB and who requested the test activity, may be involved
> in the deviation process, in case the SPR can’t be resolved.  However
> this is still under definition by the SE.  I suggest to keep the text as
> is for now.

The same complaint applies: you cannot expect product owners to assess
potentially infinte SPRs.
  
> > 2.5 Pass/Fail Criteria
> > 
> > his section should probably be rewritten in light of the procedure around
> > “deviations”, but I'm not sure what it should actually say!
> 
> The scope of the pass/fail criteria is the test case.
> The scope of a deviation is a verification element and therefore the corresponding requirement.
> I suggest to keep the definition separated.

Ok, it's fine to keep these definitions separate if they are clear. They
are very much not clear at the moment!

Is it possible to “pass” a test case (ie, have no critical SPRs) but
geenrate a deviation from the verification element?

Reply:
I created the branch u/gcomoretto/fixes. I will create a separate pull request for that to be merged in u/swinbank/proofing.

In the branch branch u/gcomoretto/fixes I corrected the following:
- renamed _product_ to _component_.
- removed the sentence on the requirements approval, this is out of scope in this document.
- removed the sentence on the synchronization back to MD
- added the link to SE confluence page
- removed footnote from section 2.1
- added few explanation on the VCD
- rephrase the test owner and tester definition
- specified that the SPRs are issues related with the software under test.
- added a reference to the deviation issues when requirements can't be fulfilled, in the pass/fails criteria.

Regarding the other pending comments:

- I did not update the table 1. The improvement is respect the previous version of the table that includes only few components (in the last issue of the document).

- We can include all components that are listed as products in LDM-294 table appendix A, but table 1 will become almost empty, and I am sure this is not good to show.

- OK, from the definition a test report is not part of the baseline, but it proves that the requirements have been successfully implemented.

- The VCD is a complex table, and it is quiet difficult to fit into it all information. Where can I found the style definition I should use?

- If there are problems in the software under test, and an SPR is opened, somebody has to to assess it.  The skills to assess the SPR can be in product owner, that in some cases may correspond to the tester,  or shared in the CCB. Do you see a better solution?